src/components/Footer/Footer.jsx

```diff
@@ -0,0 +1,45 @@
+import { useEffect, useRef, useState } from "react";
+import { useLocation } from "react-router-dom";
+import styles from "./Footer.module.css";
+
+const Footer = () => {
+    const [currentYear, setCurrentYear] = useState(new Date().getFullYear());
+    const newsletterRef = useRef(null);
+    const scriptRef = useRef(null);
+    const location = useLocation();
+
+    useEffect(() => {
+        setCurrentYear(new Date().getFullYear());
```

Comment:
I think it should be in a separate `useEffect` hook, running only on startup. You don't want to set the year every time `location.pathname` changes.